src/WpBookingsShortcode.php

```diff
@@ -1,71 +1,127 @@
+     * @param EventFactoryInterface     $eventFactory         The event factory.
+     */
+    public function __construct(
+        $key,
+        $dependencies,
+        $configFactory,
+        $containerFactory,
+        $compContainerFactory,
+        $eventManager,
+        $eventFactory
+    ) {
+        $this->_initModule($key, $dependencies, $configFactory, $containerFactory, $compContainerFactory);
+        $this->_initModuleEvents($eventManager, $eventFactory);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function setup()
+    {
+        return $this->_setupContainer($this->_loadPhpConfigFile(RC_WP_BOOKINGS_SHORTCODE_MODULE_CONFIG), [
+            /*
+             * Handles shortcode parameters before sending them to client.
+             *
+             * @since [*next-version*]
+             */
+            'eddbk_shortcode_parameters_handler' => function (ContainerInterface $c) {
+                return new ShortcodeParametersHandler(
+                    $c->get('eddbk_shortcode/edd_settings/purchase_page'),
+                    $c->get('eddbk_services_select_rm'),
+                    $c->get('sql_expression_builder'),
+                    $c->get('eddbk_admin_edit_services_ui_state_transformer')
+                );
+            },
+        ]);
+    }
+
+    /**
+     * {@inheritdoc}
+     *
+     * @since [*next-version*]
+     */
+    public function run(ContainerInterface $c = null)
+    {
+        $this->shortcodeTag = $c->get('eddbk_shortcode/shortcode_tag');
+
+        $this->_attach('eddbk_shortcode_parameters', $c->get('eddbk_shortcode_parameters_handler'));
+
+        add_shortcode($this->shortcodeTag, function ($attrs) {
+            $attrs = $attrs ? $attrs : [];
+            $attrs = $this->_trigger('eddbk_shortcode_parameters', $attrs)->getParams();
+
+            return $this->_trigger('eddbk_wizard_main_component', $attrs)->getParam('content');
+        });
+
+        $this->_attach('wp_enqueue_scripts', function () {
+            if (!$this->_shouldRenderShortcodeContent()) {
+                return;
+            }
+            $this->_trigger('eddbk_wizard_enqueue_assets');
+            $this->_trigger('eddbk_wizard_enqueue_app_state');
+        }, 999);
+
+        $this->_attach('wp_footer', function () {
+            if (!$this->_shouldRenderShortcodeContent()) {
+                return;
+            }
+            echo $this->_trigger('eddbk_wizard_components_templates', [
+                'content' => '',
+            ])->getParam('content');
+        });
+    }
+
+    /**
+     * Check that shortcode's content should be rendered.
+     *
+     * @since [*next-version*]
+     *
+     * @return bool Is shortcode's content should be rendered.
+     */
+    protected function _shouldRenderShortcodeContent()
+    {
+        return is_a(get_post(), 'WP_Post') && has_shortcode(get_post()->post_content, $this->shortcodeTag);
```

Comment:
Straight from the WP codex 😆 

- Use the more expressive `instanceof` operator.
- Re-use result of `get_post()`.
- Filter the result of this computation.